x/stake/keeper/slash.go

```diff
@@ -4,56 +4,212 @@ import (
 	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/x/stake/types"
 	"github.com/tendermint/tendermint/crypto"
 )
 
-// NOTE the current slash functionality doesn't take into consideration unbonding/rebonding records
-//      or the time of breach. This will be updated in slashing v2
-// slash a validator
-func (k Keeper) Slash(ctx sdk.Context, pubkey crypto.PubKey, height int64, fraction sdk.Rat) {
+// slash an unbonding delegation and update the pool
+func (k Keeper) slashUnbondingDelegation(ctx sdk.Context, unbondingDelegation types.UnbondingDelegation, infractionHeight int64, fraction sdk.Rat) (slashAmount sdk.Int) {
```

Comment:
can fraction be renamed to something more descriptive, perhaps `slashFactor`?

Reply:
Changed to `slashFraction`.